runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -11434,17 +11617,17 @@ J9::Power::CodeGenerator::inlineDirectCall(TR::Node *node, TR::Register *&result
          break;
 
       case TR::com_ibm_jit_JITHelpers_intrinsicIndexOfLatin1:
-         if (cg->getSupportsInlineStringIndexOf())
-            {
-            resultReg = inlineIntrinsicIndexOf(node, cg, true);
-            return true;
-            }
-         break;
-
       case TR::com_ibm_jit_JITHelpers_intrinsicIndexOfUTF16:
          if (cg->getSupportsInlineStringIndexOf())
             {
-            resultReg = inlineIntrinsicIndexOf(node, cg, false);
+            static bool disableIndexOfStringIntrinsic = feGetEnv("TR_DisableIndexOfStringIntrinsic") != NULL;
+            if (disableIndexOfStringIntrinsic)
+               return false;
```

Comment:
I'd place this directly inside the evaluator. What if someone calls the `inlineIntrinsicIndexOf_P10` function from a different place than here. We don't want to be duplicating this envvar check there as well.